### DISCUSSION TITLE
react-native-firebase support on non google service devices

### DISCUSSION THREAD

**Discussion by nadishan** [invertase/react-native-firebase#4524] [Q&A] [answered]
Hi Team,

I want to know whether this library supports non-google service devices such as Huawei. If this is supported, how do I configure this library for it?

**Comment by mikehardy** [comment, marked as answer]
Hi there! This is an area of upstream work. The best I can recommend is to check out the firebase-android-sdk repository / documentation directly. I'm not sure if we need to make changes here, but I note things like this specifically:

https://firebase.google.com/docs/android/android-play-services

However please note that some of the functionality listed there as "not requiring play services", like "auth" *do* require a very up to date version of the firebase-android-sdk:

https://firebase.google.com/support/release-notes/android#auth_v20-0-0

...and the forward port to that version is in progress but not quite completed yet - #4471 - hopefully done in a day or three

This is an area of very rapid work / progress in the underlying SDK so you'll have to bookmark that first link and the release notes link and check them during new releases to see as new services start working - where possible - without the play services library.

**Reply by edrickkochla** [reply]
Just to follow up on this discussion please...According to the installation process react native firebase requires the google services plugin, but newer huawei phones do not support google services? Could you please suggest how the react native firebase library can be implemented on new huawei phones please without using google services? I am currently using react native firebase version 6.3.4, will a version upgrade solve my issue?

**Reply by mikehardy** [reply]
@edrickkochla you have split the discussion by asking the same thing twice, both here and elsewhere, I answered you there.
